### PR TITLE
Settings UI

### PR DESCRIPTION
**Pull request by saugatmaharjan**
Tickets
- https://outsidetech.atlassian.net/browse/CP-4
- https://outsidetech.atlassian.net/browse/CP-5

- [x] add auto response option select
- [x] add response text field
- [x] add react hook form
- [x] add working hours ui
- [x] maintain order of days
- [x] watch auto response settings change to manage form data
- [x] fix ui according to figma design
- [x] save api integration

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,6 +29,7 @@
     "eslint": "latest",
     "eslint-config-next": "latest",
     "postcss": "latest",
+    "prisma": "^5.4.2",
```
Comment (sazanrjb) [inline package.json:32]: prisma is already in dependencies!

File: src/app/page.tsx [hunk 2]
```diff
@@ -1,136 +1,90 @@
...
+import { SettingResponse } from "@/types/setting";
+import { $Enums } from "@prisma/client";
 
-type SearchParams = { [key: string]: string | string[] | undefined }
+type SearchParams = { [key: string]: string | string[] | undefined };
+
+const settingsService = new SettingService();
 
 async function getContent(searchParams: SearchParams) {
   if (!process.env.COPILOT_API_KEY) {
```
Comment (sazanrjb) [inline src/app/page.tsx:19]: we can use this from app config

Comment (sazanrjb) [reply src/app/page.tsx:19]: oh.. the branch hasnt merged though

File: src/constants/index.ts [hunk 3]
```diff
@@ -0,0 +1,199 @@
+import { SelectOption } from "@/app/components/Select";
+import { $Enums } from "@prisma/client";
+
+export enum AUTO_RESPONSE {
```
Comment (sazanrjb) [inline src/constants/index.ts:4]: is this needed? we have SettingType enum

File: src/constants/index.ts [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+export const AUTO_RESPONSE_OPTIONS: SelectOption<$Enums.SettingType>[] = [
+  {
+    value: $Enums.SettingType.OUTSIDE_WORKING_HOURS,
+    label: "Outside working hours",
+  },
+  { value: $Enums.SettingType.ENABLED, label: "Always on" },
+  { value: $Enums.SettingType.DISABLED, label: "Off" },
+];
+
+export const DAYS = {
```
Comment (sazanrjb) [inline src/constants/index.ts:19]: we can use DayOfWeek instead of this

File: src/constants/index.ts [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+export type DAY_VALUE = (typeof DAYS)[keyof typeof DAYS];
+export type DAY_KEY = keyof typeof DAYS;
+
+export interface SelectedDay {
+  day: DAY_VALUE;
+  startHour: HOUR;
+  endHour: HOUR;
+}
+
+export enum TIMEZONE {
```
Comment (sazanrjb) [inline src/constants/index.ts:75]: they have suggested to use react timezone library, can you check the message

**Review by sazanrjb — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (saugatmaharjan) [reply package.json:36]: done

**Comment by saugatmaharjan** [comment]
will fix the comments in separate PR

**Pull request merged**